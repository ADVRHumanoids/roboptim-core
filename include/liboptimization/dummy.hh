// Copyright (C) 2009 by Thomas Moulard, FIXME.
//
// This file is part of the liboptimization.
//
// liboptimization is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// liboptimization is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with liboptimization.  If not, see <http://www.gnu.org/licenses/>.


/**
 * \brief Implementation of the dummy module (always fail).
 */

#ifndef OPTIMIZATION_DUMMY_HH
# define OPTIMIZATION_DUMMY_HH
# include <liboptimization/solver.hh>

namespace optimization
{
  /// Dummy solver which always fails.
  class DummySolver : public Solver<Function, Function>
  {
  public:
    /// Define parent's type.
    typedef Solver<Function, Function> parent_t;

    /// Constructor.
    explicit DummySolver (const problem_t&) throw ();
    /// Destructor.
    virtual ~DummySolver () throw ();

    /// Implement the solve algorithm.
    virtual void solve () throw ();
  };

}; // end of namespace optimization

#endif //! OPTIMIZATION_DUMMY_HH
