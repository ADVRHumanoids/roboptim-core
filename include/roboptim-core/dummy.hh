// Copyright (C) 2009 by Thomas Moulard, AIST, CNRS, INRIA.
//
// This file is part of the roboptim.
//
// roboptim is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// roboptim is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with roboptim.  If not, see <http://www.gnu.org/licenses/>.


/**
 * \brief Implementation of the dummy module (always fail).
 */

#ifndef ROBOPTIM_CORE_DUMMY_HH
# define ROBOPTIM_CORE_DUMMY_HH
# include <roboptim-core/solver.hh>

namespace roboptim
{
  /// Dummy solver which always fails.
  class DummySolver : public Solver<Function, const Function*>
  {
  public:
    /// Define parent's type.
    typedef Solver<Function, const Function*> parent_t;

    /// Constructor.
    explicit DummySolver (const problem_t&) throw ();
    /// Destructor.
    virtual ~DummySolver () throw ();

    /// Implement the solve algorithm.
    virtual void solve () throw ();
  };

} // end of namespace roboptim

#endif //! ROBOPTIM_CORE_DUMMY_HH
