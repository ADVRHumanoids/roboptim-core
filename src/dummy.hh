// Copyright (C) 2009 by Thomas Moulard, FIXME.
//
// This file is part of the liboptimization.
//
// liboptimization is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// liboptimization is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with liboptimization.  If not, see <http://www.gnu.org/licenses/>.


/**
 * \file src/dummy.hh
 *
 * \brief Implementation of the dummy module (always fail).
 */

#ifndef OPTIMIZATION_DUMMY_HH
# define OPTIMIZATION_DUMMY_HH
# include <solver.hh>

namespace optimization
{
  /// Dummy solver which always fails.
  class DummySolver : public Solver
  {
  public:
    /// Constructor.
    explicit DummySolver (const Problem&) throw ();
    /// Destructor.
    virtual ~DummySolver () throw ();
    /// Return the default solver error.
    virtual result_t getMinimum () throw ();
  };

} // end of namespace optimization

#endif //! OPTIMIZATION_DUMMY_HH
