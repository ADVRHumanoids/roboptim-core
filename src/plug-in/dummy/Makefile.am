include $(top_srcdir)/build-aux/init.mk

lib_LTLIBRARIES = liboptimization-dummy.la

liboptimization_dummy_la_SOURCES = dummy.cc
liboptimization_dummy_la_LDFLAGS = -module
