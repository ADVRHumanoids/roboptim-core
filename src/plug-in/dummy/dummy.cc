// Copyright (C) 2009 by Thomas Moulard, FIXME.
//
// This file is part of the liboptimization.
//
// liboptimization is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// liboptimization is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with liboptimization.  If not, see <http://www.gnu.org/licenses/>.


/**
 * \file dummy.cc
 *
 * \brief Implementation of the dummy module (always fail).
 */

#include <solver.hh>

namespace optimization
{

  class DummySolver : public Solver
  {
  public:
    DummySolver (const function_t& fct) throw ()
      : Solver (fct)
    {
    }

    virtual ~DummySolver () throw ()
    {
    }

    virtual result_t
    getMinimum () throw ()
    {
      return result_t (SolverError ());
    }
  };

} // end of namespace optimization


extern "C"
{
  optimization::Solver*
  create (const optimization::DummySolver::function_t& arg)
  {
    return new optimization::DummySolver (arg);
  }

  void
  destroy (optimization::Solver* p)
  {
    delete p;
  }

  unsigned
  version ()
  {
    return optimization::Solver::PLUGIN_INTERFACE_VERSION;
  }
}
