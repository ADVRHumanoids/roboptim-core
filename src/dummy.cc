// Copyright (C) 2009 by Thomas Moulard, FIXME.
//
// This file is part of the liboptimization.
//
// liboptimization is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// liboptimization is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with liboptimization.  If not, see <http://www.gnu.org/licenses/>.


/**
 * \file src/dummy.cc
 *
 * \brief Implementation of the dummy module (always fail).
 */

#include "dummy.hh"

namespace optimization
{
  DummySolver::DummySolver (function_t fct,
                            size_type n,
                            gradient_t g,
                            hessian_t h,
                            jacobian_t j) throw ()
    : Solver (fct, n, g, h, j)
  {
    result_ = SolverError ("The dummy solver always fail.");
  }

  DummySolver::~DummySolver () throw ()
  {
  }

  DummySolver::result_t
  DummySolver::getMinimum () throw ()
  {
    return result_;
  }

} // end of namespace optimization
