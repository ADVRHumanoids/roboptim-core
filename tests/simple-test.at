# Copyright (C) 2009 by Thomas Moulard, FIXME.                    -*-Autotest-*-
#
# This file is part of the liboptimization.
#
# liboptimization is free software: you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# liboptimization is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with liboptimization.  If not, see <http://www.gnu.org/licenses/>.

AT_SETUP([Check DummySolver.])

AT_TESTED([simple-test])

cp $abs_srcdir/simple-test.stdout expout
AT_CHECK([$CHECK_PREFIX `which simple-test`],
         [0], [expout], [ignore])

AT_CLEANUP

# End of simple-test.at
