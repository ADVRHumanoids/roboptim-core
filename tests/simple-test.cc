// Copyright (C) 2009 by Thomas Moulard, FIXME.
//
// This file is part of the liboptimization.
//
// liboptimization is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// liboptimization is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with liboptimization.  If not, see <http://www.gnu.org/licenses/>.

#include <iostream>
#include <boost/lambda/lambda.hpp>
#include <boost/variant/get.hpp>

#include "common.hh"
#include <dummy.hh>

using namespace optimization;
using namespace boost;
using namespace boost::lambda;

typedef DummySolver solver_t;

double my_fun (const solver_t::array_t& x)
{
  return x[0] * x[3] * (x[0] + x[1] + x[2]) + x[3];
}

int run_test ()
{

  // Check with identity function (fun x -> x).
  {
    solver_t solver (ret<const double&> (_1[0]), 1);
    solver_t::result_t res = solver.getMinimum ();
    boost::get<SolverError> (res);
  }

  // Check with basic function (fun x y -> x + y).
  {
    solver_t solver (ret<const double&> (_1[0]) + ret<const double&> (_1[1]), 2);
    solver_t::result_t res = solver.getMinimum ();
    boost::get<SolverError> (res);
  }

  // Check with complex function.
  {
    solver_t solver (my_fun, 4, solver_t::gradient_t ());
    solver_t::result_t res = solver.getMinimum ();
    boost::get<SolverError> (res);
  }

  return 0;
}

GENERATE_TEST ()
