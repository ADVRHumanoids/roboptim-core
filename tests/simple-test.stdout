Function
Result bound: (-inf, inf)
Arguments bounds (-inf, inf), (-inf, inf), (-inf, inf), (-inf, inf)
Result scale: 1
Arguments scale: 1, 1, 1, 1
No constraints
No starting point.
---
Function
Result bound: (-inf, inf)
Arguments bounds (-inf, inf), (-inf, inf), (-inf, inf), (-inf, inf)
Result scale: 1
Arguments scale: 1, 1, 1, 1
No constraints
No starting point.
